### PR TITLE
Unit test messaging providers endpoints

### PR DESCRIPTION
**Pull request by Thurfjell**
### Ticket:

- [20683](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/20683)

### Description

Unit test erroruos endpoints sanitation for /api/v1/providers path.

Using mostly following pattern, making sure following scenarios fail
- field missing
- field empty (for string)
- field null type (and we assume this will work for the other valid json types, number, object, array, boolean, string)

There are also search query validation where appropriate as well as url parameter.

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [ ] **New feature**
- [x] **Dev change**
- [ ] **Additional tests**
- [ ] **Documentation**
- [ ] **Other**

### Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

File: apps/messaging-api/src/app.ts [hunk 1]
```diff
@@ -19,14 +19,24 @@ import fastifyMultipart from "@fastify/multipart";
...
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = dirname(__filename);
 
 dotenv.config();
 
 export async function build(opts?: FastifyServerOptions) {
-  const app = fastify(opts).withTypeProvider<TypeBoxTypeProvider>();
+  const app = fastify({
+    ...opts,
+    ajv: {
```
Comment (alfonsograziano) [inline apps/messaging-api/src/app.ts:33]: @peschina shall we apply the same in payments?

Comment (Thurfjell) [reply apps/messaging-api/src/app.ts:33]: Good that you raise it. @alfonsograziano 

I would recommend that we apply this rule on all APIs.

**Review by rgrnearform — APPROVED** [review]
lgtm

File: apps/messaging-api/src/app.ts (hunk 1 above)
Comment (alfonsograziano) [reply apps/messaging-api/src/app.ts:33]: Agreed :)

Comment (peschina) [reply apps/messaging-api/src/app.ts:33]: yes!

**Pull request merged**